README.md

```diff
@@ -30,13 +30,18 @@ A `fetch` is a function with the following signature:
 fetch: A -> Promise[P]
 ```
 
-and we write `fetch: A ~> P` for short.
+whenever we write `~>` instead of `->` we're implying `Promise[]`
+
+ex: we can write `fetch: A ~> P` for short.
 
 ## Classification
 
-Given a fetch `fetch: A -> Promise[P]`:
+Given a fetch `fetch: A ~> P`:
 
-- a *catalog* for `fetch`
+- an *index* for `fetch` is defined as `List[A] ~> List[P]`
+  - ex: `getListOfThingsByIds`
+- a *catalog* for `fetch` is defined as `S ~> List[P]`
+  - ex: `searchThingsWithQuery`
```

Comment:
also, not sure about introducing `S` instead of `A`, but in any case `A` (or `S`) should be the same input both for catalog and index